File: java/src/main/java/command/Transfer.java
package command;

import java.util.function.Consumer;

public class Transfer {
    private final Consumer<TransferParam> command;

    public Transfer(Consumer<TransferParam> command) {
        this.command = command;
    }

    public void start(TransferParam transferParam) {
        command.accept(transferParam);
    }
}

File: command/transfer.go
package command

type transfer struct {
	command func(transferParam TransferParam)
}

func NewTransfer(command func(transferParam TransferParam)) *transfer {
	return &transfer{command: command}
}

func (t transfer) Start(transferParam TransferParam) {
	t.command(transferParam)
}

type TransferParam struct {
	amount  float64
	account string
}

func NewTransferParam(amount float64, account string) TransferParam {
	return TransferParam{amount: amount, account: account}
}
